### ISSUE TITLE
{failed_to_start_child,syslog_monitor,noproc} on Erlang 21

### ISSUE THREAD

**Issue by gotthardp** [schlagert/syslog#15]
Hello. Start of the syslog application fails on Erlang 21. Even the eunit tests fail with Erlang 21 (and succeed with all previous versions). I am getting the following error:
```
=CRASH REPORT==== 27-Jul-2018::20:29:08.138859 ===
  crasher:
    initial call: application_master:init/4
    pid: <0.305.0>
    registered_name: []
    exception exit: {{shutdown,{failed_to_start_child,syslog_monitor,noproc}},
                     {syslog,start,[normal,[]]}}
      in function  application_master:init/4 (application_master.erl, line 138)
    ancestors: [<0.304.0>]
    message_queue_len: 1
    messages: [{'EXIT',<0.306.0>,normal}]
    links: [<0.304.0>,<0.42.0>]
    dictionary: []
    trap_exit: true
    status: running
    heap_size: 376
    stack_size: 27
    reductions: 193
  neighbours:
```
Could you please have a look on what could be wrong?

**Comment by schlagert** [comment]
I'm sorry, I have to admit, that due to the introduction of the new logger API, syslog is not (yet) compatible with OTP 21.

Additionally, making it compatible is not trivial and I don't yet know when I will have the time to make the necessary changes.

**Comment by gotthardp** [comment]
Thank you for the information.

**Comment by schlagert** [comment]
Ok, after trying to fix the unit tests with OTP 21 for several hours I have given up on them...

However, `syslog` should now play together with OTP 21. Please note, that the `logger_sasl_compatible` kernel configuration is vital. Additionally, the integration will still use the `error_logger`, not the new `logger` API. I will leave this ticket open, until I find the time to implement a _proper_ solution.

**Comment by schlagert** [comment]
I finally found the time and have put in the effort for proper OTP 21 support. It can be tested with `master` and will be released with `4.0.0`.

**Comment by gotthardp** [comment]
Thanks a lot!
